tools/py/mux-positions.py

```diff
@@ -4,21 +4,61 @@
 
 import os
 import sys
+import numpy as np
 import argparse
 from ipi.utils.messages import verbosity
+import copy
 
 from ipi.utils.io import open_backup, iter_file_name, print_file
 
 
 description = """
 Read positions of individual beads from a set of trajectory files and
-multiplex them into a single output trajectory. Trajectory file formats are
-inferred from file extensions, the number of beads is given by the number of
-input files.
+a) multiplex them into a single output trajectory
+b) wrap/unwrap them.
+Trajectory file formats are inferred from file extensions, the number
+of beads is given by the number of input files.
 """
 
-
-def main(fns_in, fn_out, begin, end, stride):
+def cell_dot_pos(cell,pos):
+    s = np.array([])
+    for row in pos:
+        s = np.append(s,np.dot(cell,row))
+    s.shape = (len(pos),3)
+    return s
+
+def wrap_positions(frame):
+    pos = frame['atoms'].q.copy()
+    pos.shape = (len(pos)/3,3)
+    cell = frame['cell'].h
+    cell_inv = np.linalg.inv(cell)
+    s = cell_dot_pos(cell_inv,pos)
+    sc = s - np.round(s)
+    frame['atoms'].q = cell_dot_pos(cell,sc).flatten()
+    return frame
+
+
+def unwrap_positions(frame,framelast):
+    frame_wr = wrap_positions(frame)
```

Comment:
Why do you wrap before unwrapping?

Reply:
did remove the wrap before unwrapping. Also, put wrap/unwrap to the bottom. Created separate functions for wrap unwrap due to your suggestion above: 
if wrap:
    wrap positions
if unwrap:
    unwraps positions